tests/source/issue_5721.rs

```diff
@@ -0,0 +1,8 @@
+#![feature(non_lifetime_binders)]
+#![allow(incomplete_features)]
+
+trait Other<U: ?Sized> {}
+
+trait Trait<U> 
+where
+    for<T> U: Other<T> {}
```

Comment:
No real issue keeping the `source` file, but I just want to point out that the `target` file is enough to show that the input is idempotent, and that rustfmt no longer remove the `for<T>`.